### ISSUE TITLE
Group annotations into one unique location if their distance is within a custom radius distance

### ISSUE THREAD

**Issue by kuoliangkwong** [choefele/CCHMapClusterController#77]
Hi,

Is there any way to make annotations to be grouped as one unique location if they are, say, within 50m in radius? I tried using the cellSize but it seems like it's used to clustering annotations.

**Comment by choefele** [comment]
You are right about the cell size.

Locations are considered the same if [their geohash is the same](https://github.com/choefele/CCHMapClusterController/blob/master/CCHMapClusterController/CCHMapClusterControllerUtils.m#L291). [More about geohashes](https://en.wikipedia.org/wiki/Geohash). The code is currently using a geohash length of 9.

While you could make the geohash length a property of the cluster controller, I'm not sure there's an easy way to turn distance in meters into a length value. Otherwise, the geohash length is a bit opaque.

Perhaps you can pre-process your data to cut off significant bits of the coordinate to achieve the same? You could even do this on the server thus saving time on the client.

**Comment by kuoliangkwong** [comment]
Is it the one from <a href="https://github.com/choefele/CCHMapClusterController/blob/master/CCHMapClusterController/CCHMapClusterControllerUtils.m#L33">here<a>? I guess we could just add a property for geohash length. Is it ok we can pull request once we done adding the property?

I tried cutting off the significant bits by rounding down the latitude and longitude to 3 decimal points, but it seems like the location is slightly shifted.

**Comment by choefele** [comment]
Yes, that's what I meant. `CCHMapClusterOperation` uses [`anyObject`](https://github.com/choefele/CCHMapClusterController/blob/master/CCHMapClusterController/CCHMapClusterOperation.m#L138) to display unique locations. Basically you'll get a random location from the set.

I'm thinking you might be best off with preprocessing your data so you can define which location is used for nearby locations.
